### ISSUE TITLE
tokio 1.0

### ISSUE THREAD

**Issue by clux** [clux/kube-rs#339]
Just a tracking issue for the tokio 0.3 upgrade.

A couple of things needed;

- [x] minor changes in [tokio::time](https://docs.rs/tokio/0.3.3/tokio/time/index.html) for old runtime (if we care anymore)
- [ ] `reqwest` needs to bump it https://github.com/seanmonstar/reqwest/issues/1060 (it's in their master, you can use patch (see below), but a kube release blocked on a crates release of `reqwest`)

**Comment by jmjoy** [comment]
Maybe wait tokio 1.0? Because:
> The Tokio team plans to release Tokio 1.0 by the end of December 2020.

https://tokio.rs/blog/2020-10-tokio-0-3

**Comment by clux** [comment]
Oh, if that's true, then might as well wait with releasing a new version that depends on a new tokio.

However; if the point of a beta is that the ecosystem has some time to test it out, then maybe that date will be delayed? Reqwest and others haven't caught up to the beta yet.

**Comment by teozkr** [comment]
Tokio 1.0 [is out](https://tokio.rs/blog/2020-12-tokio-1-0), so it 0.3 seems a bit moot now.. :P

**Comment by teozkr** [comment]
Progress tracker:

- [ ] reqwest
- [X] hyper - 0.14.2
- [ ] hyper-tls - merged but unreleased, https://github.com/hyperium/hyper-tls/pull/79#event-4155650741
- [X] tokio-tls
- [X] h2
- [ ] kube
- [ ] kube-runtime

**Comment by clux** [comment]
i'll keep this open until we do a release with it in (blocked by reqwest making a release)

**Comment by jbg** [comment]
For anyone wanting to use a git dependency on `kube` to try this out, note that dependency patching only works in the root crate of a dependency tree, so you'll need to copy the patch from kube's Cargo.toml into yours:

```
[patch.crates-io]
# https://github.com/seanmonstar/reqwest/pull/1076 hasn't been released yet
reqwest = { git = "https://github.com/seanmonstar/reqwest.git", rev = "1f425a0244bcd7b4565dceb9076450d951f2ec03" }
```

**Comment by kazk** [comment]
`reqwest` v0.11.0 was released: https://github.com/seanmonstar/reqwest/pull/1133

**Comment by clux** [comment]
published in kube* `0.47.0`
